lib/ex_doc/formatter/html.ex

```diff
@@ -390,6 +384,36 @@ defmodule ExDoc.Formatter.HTML do
     |> Enum.sort_by(fn extra -> GroupMatcher.group_index(groups, extra.group) end)
   end
 
+  def generate_redirects(config, ext) do
+    config.redirects
+    |> Map.put_new("index", config.main)
+    |> Enum.map(fn {from, to} ->
+      source = stringify_redirect_item(from) <> ext
+      destination = stringify_redirect_item(to) <> ext
+      generate_redirect(source, config, destination)
+
+      source
+    end)
+  end
+
+  defp stringify_redirect_item(item) when is_binary(item) do
+    item
+  end
+
+  defp stringify_redirect_item(item) when is_atom(item) do
+    inspected = inspect(item)
+
+    case to_string(item) do
+      "Elixir." <> ^inspected -> inspected
+      other -> other
+    end
+  end
+
+  defp stringify_redirect_item(item) do
+    raise ArgumentError,
+          "Redirect source and destination must be a string or an atom, got: #{inspect(item)}"
```

Comment:
Error messages typically start in lowercase:

```suggestion
          "redirect source and destination must be a string or an atom, got: #{inspect(item)}"
```